auto_nag/bzcleaner.py

```diff
@@ -219,6 +233,84 @@ def add_auto_ni(self, bugid, data):
             }
         return True
 
+    def add_prioritized_action(self, bug, quota_name, needinfo=None, autofix=None):
+        """
+        - `quota_name` is the key used to apply the limits, e.g., triage owner, team, or component
+        """
+        assert needinfo or autofix
+
+        # Avoid having more than one ni from our bot
+        if needinfo and self.has_bot_set_ni(bug):
+            needinfo = autofix = None
+
+        action = {
+            "bugid": str(bug["id"]),
+            "needinfo": needinfo,
+            "autofix": autofix,
+        }
+
+        sort_key = self.get_bug_sort_key(bug)
+        if sort_key is not None:
+            action["key"] = sort_key
+
+        self.quota_actions[quota_name].append(action)
+
+    def get_bug_sort_key(self, bug):
+        return None
+
+    def _populate_prioritized_actions(self, bugs):
+        max_actions = self.get_max_actions()
+        max_ni = self.get_max_ni()
+        exclude_no_action_bugs = (
+            len(self.quota_actions) > 0 and self.exclude_no_action_bugs()
+        )
+        bugs_with_action = set()
+
+        for actions in self.quota_actions.values():
+            if len(actions) > max_ni or len(actions) > max_actions:
+                actions.sort(
+                    key=lambda x: (not x["needinfo"], x["key"])
+                    if "key" in x
+                    else not x["needinfo"]
+                )
+
+            ni_count = 0
+            actions_count = 0
+            for action in actions:
+                bugid = action["bugid"]
+                if max_actions > 0 and actions_count >= max_actions:
+                    break
+
+                if action["needinfo"]:
+                    if max_ni > 0 and ni_count >= max_ni:
+                        continue
+
+                    ok = self.add_auto_ni(bugid, action["needinfo"])
+                    if not ok:
+                        # If we can't needinfo, we do add the autofix
+                        continue
+
+                    if "extra" in action["needinfo"]:
+                        self.extra_ni[bugid] = action["needinfo"]["extra"]
+
+                    if exclude_no_action_bugs:
```

Comment:
Nit: this check could be removed because you are only using `bugs_with_actions` if `exclude_no_action_bugs` is True later on. You could always add the bug IDs to the set no matter what

Reply:
No problem. Done in 1c43ee7e2d1771d6741f01b55e6b3d5ac4e7d91a.